packages/contracts-bedrock/scripts/checks/interfaces/main.go

```diff
@@ -34,6 +34,85 @@ var excludeContracts = []string{
 	"KontrolCheatsBase", "IResolvedDelegateProxy",
 }
 
+// Contracts that don't need an interface
+var excludeSourceContracts = []string{
+	// Libraries don't need interfaces
+	"Bytes", "Bytes32", "Encoding", "Hashing", "LibAddress", "LibBytesUtils", "LibMap", "LibMath", "LibSort", "LibString",
+
+	// Test helpers and mocks
+	"ERC20", "WETH9", "TestUtil", "TestERC20", "TestLib", "MockOVMCodeChecker", "MockERC20",
+	"MIPS", "MIPSMemory", "PreimageOracle", "PreimageKeyLib",
+
+	// Abstract contracts
+	"CrossDomainMessenger", "Semver", "FeeVault", "StandardBridge", "OptimismPortal",
+
+	// Special exclusions for specific reasons
+	"SafeCall", "ResourceMetering", "Transactor", "AddressManager", "AddressResolver",
+	"L1Block", "GasPriceOracle", "Burn",
+	"AutoRedeem", "Initializable", "Predeploys", "Proxy",
+	"SchemaRegistry", "AttestationStation", "EAS",
+
+	// These are purely implementation contracts without interfaces needed
+	"Ownable", "UUPSUpgradeable", "Clones", "Create2",
+	"ProxyAdmin", "L1ChugSplashProxy", "ERC721Bridge", "ERC721BridgeLegacy",
+	"CallContextHelper", "ImmutableProxy", "ResolvedDelegateProxy",
+
+	// L1 contracts
+	// NOTE: The following contracts should have interfaces and are intentionally NOT excluded:
+	// - ETHLockbox
+	// - DataAvailabilityChallenge
+	// - L1CrossDomainMessenger
+	// - L1ERC721Bridge
+	// - L1StandardBridge
+	"OPContractsManagerContractsContainer", "OPContractsManagerBase",
+	"OPContractsManagerGameTypeAdder", "OPContractsManagerUpgrader", "OPContractsManagerDeployer",
+	"OPContractsManagerInteropMigrator", "OPContractsManager", "OptimismPortal2",
+	"ProtocolVersions", "ProxyAdminOwnedBase", "StandardValidatorBase", "StandardValidatorV180",
+	"StandardValidatorV200", "StandardValidatorV300", "SuperchainConfig", "SystemConfig",
+
+	// L2 contracts
+	"BaseFeeVault", "CrossDomainOwnable", "CrossDomainOwnable2", "CrossDomainOwnable3",
+	"CrossL2Inbox", "ETHLiquidity", "L1FeeVault", "L2CrossDomainMessenger", "L2ERC721Bridge",
+	"L2StandardBridge", "L2StandardBridgeInterop", "L2ToL1MessagePasser", "L2ToL2CrossDomainMessenger",
+	"OperatorFeeVault", "OptimismMintableERC721", "OptimismMintableERC721Factory",
+	"OptimismSuperchainERC20", "OptimismSuperchainERC20Beacon", "OptimismSuperchainERC20Factory",
+	"SequencerFeeVault", "SuperchainERC20", "SuperchainTokenBridge", "SuperchainWETH",
+
+	// Cannon contracts
+	"MIPS2", "MIPS64",
```

Comment:
There are some more contracts here that should have interfaces but I'm happy to merge this and I'll clean up the exclusion list later